### PR TITLE
Update primegen recipe

### PR DESCRIPTION
**Pull request by ohnx**
Re: https://github.com/haikuports/haikuports/pull/1812#discussion_r154351195

### CODE REVIEW

File: sci-libs/primegen/primegen-0.97.recipe [hunk 1]
```diff
@@ -43,6 +43,11 @@ BUILD_REQUIRES="
 	haiku_devel
 	"
 
+TEST_PREREQUIRES="
+	cmd:md5sum
+	"
+BUILD_PREREQUIRES += TEST_PREREQUIRES
```
Comment (fbrosson) [inline sci-libs/primegen/primegen-0.97.recipe]: I guess you meant
```
BUILD_PREREQUIRES+="$TEST_PREREQUIRES"
```

Comment (ohnx) [reply sci-libs/primegen/primegen-0.97.recipe]: Sorry, I should have properly checked...

Comment (fbrosson) [reply sci-libs/primegen/primegen-0.97.recipe]: No problem  ;-)

File: sci-libs/primegen/primegen-0.97.recipe [hunk 2]
```diff
@@ -43,6 +43,11 @@ BUILD_REQUIRES="
 	haiku_devel
 	"
 
+TEST_PREREQUIRES="
+	cmd:md5sum
+	"
+BUILD_PREREQUIRES+="$TEST_PREREQUIRES"
```
Comment (fbrosson) [inline sci-libs/primegen/primegen-0.97.recipe]: On the GCI site, in a reply to another student, PulkoMandy wrote:
> Eventually we will change the tool to allow a TEST_REQUIRES or similar, but this hasn't been done yet.

Later, in PR 1812, you and him suggested using TEST_PREREQUIRES.

@pulkomandy (and others), which one do you prefer? TEST_REQUIRES or TEST_**PRE**REQUIRES ?
Shall we make an online poll using GitHub's thumbs (up and/or down) on any of the next two comments?

Comment (fbrosson) [reply sci-libs/primegen/primegen-0.97.recipe]: TEST_PREREQUIRES

Comment (fbrosson) [reply sci-libs/primegen/primegen-0.97.recipe]: TEST_REQUIRES

Comment (waddlesplash) [reply sci-libs/primegen/primegen-0.97.recipe]: I'm still not sure what the distinction was created for in the first place; someone would have to investigate in HaikuPorter. Perhaps @mmlr knows?

Comment (pulkomandy) [reply sci-libs/primegen/primegen-0.97.recipe]: The distinction is relevant only when cross compiling (bootstrap). pre-requires are for running on the host, whereas requires are built for the target. In the case of tests, we can only run them on target, so the distinction is not useful.

File: sci-libs/primegen/primegen-0.97.recipe [hunk 3]
```diff
@@ -43,10 +43,10 @@ BUILD_REQUIRES="
 	haiku_devel
 	"
 
-TEST_PREREQUIRES="
+TEST_REQUIRES="
 	cmd:md5sum
 	"
-BUILD_PREREQUIRES+="$TEST_PREREQUIRES"
+BUILD_REQUIRES+="$TEST_REQUIRES"
```
Comment (fbrosson) [inline sci-libs/primegen/primegen-0.97.recipe]: Oops, you accidentally also remove the "PRE" in BUILD_PREREQUIRES.

Comment (ohnx) [reply sci-libs/primegen/primegen-0.97.recipe]: @pulkomandy said:
> The distinction is relevant only when cross compiling (bootstrap). pre-requires are for running on the host, whereas requires are built for the target. In the case of tests, we can only run them on target, so the distinction is not useful.

I assume then, we would use REQUIRE's?

Comment (waddlesplash) [reply sci-libs/primegen/primegen-0.97.recipe]: Indeed.

**Comment by fbrosson** [comment]
Now that this PR has been merged, the branch it is based on is no longer needed.
You should therefore click "Delete branch" for this PR to get rid of an unneeded branch in your repo. Thanks!
GitHub help: [Deleting unused branches](https://help.github.com/articles/deleting-unused-branches/)

**Comment by ohnx** [comment]
Thanks @fbrosson  :)

**Pull request merged**